code/Gerrit.md

```diff
@@ -0,0 +1,282 @@
+* ...
+
+Two people approved your PR in Github, and build passed, but some new changes have been committed to the `main` branch since you originally created the PR, so you need to merge the main branch and update the PR.
+
+```
+git pull --rebase origin main
+git push gerrit HEAD:refs/for/main
+```
+
+Now your commit history looks like this:
+
+* file.go: fixed a bug
+* the latest commit from the remote `main` branch
+* ...
+
+Build passes, and with your two approvals, submit to `main`. You do so
+
+Now, the commit history of the `main` branch looks like this:
+
+* file.go: fixed a bug
+* the previous commit on the `main` branch
+* ...
+
+## Advantages (and disadvantages) of Gerrit
+
+Gerrit essentially requires that the commit you are submitting for review is _exactly_ the commit being submitted to the `main` branch, including the commit message and parent commit. By contrast, Github only requires that the files changed are approved. This means that any change that is submitted to the `main` branch via Gerrit must have a commit history that looks exactly like:
+
+* The commit for the change
+* The latest commit in the `main` branch
+
+Which is conveniently exactly what the `main`branch will look like after the new change is submitted.
+
+One of the things this gives us is the ability to review commit messages in Gerrit, which is important since we have some [things to keep in mind](./Git.md) regarding commit messages, and it is easy to make a mistake when merging a Github pull request.
+
+Another advantage of this is it means that you can track multiple reviews on the same branch using Gerrit, which is not possible in Github.
+
+In Github, if you want to get concurrent reviews on two changes, A and B, where B is dependent on A, then you will need to make two branches:
+
+`branch-a`
+
+* A commits
+* main branch commits
+
+`branch-b`
+
+* B commits
+* A commits
+* main branch commits
+
+meaning that a pull request to merge `branch-b` into `main` will _also_ contain the changes from `branch-a`, making the review more complicated.
+
+In Gerrit, you can have one branch with two commits:
+
+* single commit for B
+* single commit for A
+* main branch commits
+
+Now, when you push to Gerrit, a separate review will be created for A and B, and each review only contains the changes for a single commit. However, the B commit will still not be submittable until the A commit is submittable (or has already been submitted).
+
+One of the most common issues people have with Gerrit is getting used to an unfamiliar git workflow. Once you get some practice with it, you will be able to decide whether or not you prefer it. The advantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits. The disadvantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits. 
```

Comment:
This paragraph contains  the same sentences as an advantage and a disadvantage

> The advantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits.

> The disadvantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits.

Reply:
yeah that was intentional and I was attempting some humor :joy: but maybe it would be better to have different sentences